Src/LiquidProjections.NHibernate/NHibernateProjector.cs

```diff
@@ -119,8 +119,13 @@ public ShouldRetry ShouldRetry
             {
                 throw new ArgumentNullException(nameof(transactions));
             }
+            
+            long? lastCheckpoint = GetLastCheckpoint();
+            IEnumerable<IList<Transaction>> transactionBatches = transactions
+                .Where(t => t.Checkpoint > lastCheckpoint.GetValueOrDefault(-1))

```

Comment:
I would just check `lastCheckpoint.HasValue` here instead of the not-so-clear `-1` constant default